src/lib.rs

```diff
@@ -1147,8 +1150,128 @@ pub enum Expression {
     ArrayLength(Handle<Expression>),
 }
 
-/// A code block is just a vector of statements.
-pub type Block = Vec<Statement>;
+/// A code block is a vector of statements, with maybe a vector of spans.
+#[derive(Debug, Clone, Default)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize))]
+#[cfg_attr(
+    any(feature = "serialize", feature = "deserialize"),
+    serde(transparent)
+)]
+pub struct Block {
+    body: Vec<Statement>,
+    #[cfg(feature = "span")]
+    #[cfg_attr(any(feature = "serialize", feature = "deserialize"), serde(skip))]
+    span_info: Vec<Span>,
+}
+
+impl Block {
+    pub fn new() -> Self {
+        Self {
+            body: Vec::new(),
+            #[cfg(feature = "span")]
+            span_info: Vec::new(),
+        }
+    }
+
+    pub fn from_vec(body: Vec<Statement>) -> Self {
+        #[cfg(feature = "span")]
+        let span_info = std::iter::repeat(Span::Unknown).take(body.len()).collect();
+        Self {
+            body,
+            #[cfg(feature = "span")]
+            span_info,
+        }
+    }
+
+    pub fn body(&self) -> &Vec<Statement> {
```

Comment:
nit: any reason this doesn't return `&[Statement]`?

Reply:
Oh that's just bad anyway.